.github/workflows/build-and-test.yml

```diff
@@ -82,9 +87,34 @@ jobs:
                   python -m pip install --upgrade pip nose setuptools
                   python -m pip install -r requirements.txt -r requirements-dev.txt
                   python setup.py install
+                  echo ::set-env name=PYTHON::$(where python)
+
+            - name: Set up MacOS python
+              if: matrix.os != 'windows-latest'
+              uses: goanpeca/setup-miniconda@v1.1.2
+              with:
+                  update-conda: false
+                  activate-environment: env
+                  auto-update-conda: true 
+                  python-version: ${{ matrix.python-version }}
+                  conda-channels: defaults
+                  channels: conda-forge, anaconda
+            
+            - name: Install dependencies MacOS
+              if: matrix.os != 'windows-latest'
+              shell: bash -l {0}
+              run: |
+                  conda upgrade -y pip setuptools
+                  conda install Cython requests numpy
+                  python ./scripts/convert-requirements-to-conda-yml.py requirements.txt requirements-dev.txt requirements-gui.txt > requirements-all.yml
+                  conda env update --file requirements-all.yml
+                  python setup.py install
+                  echo ::set-env name=PYTHON::$(which python)
 
             - name: Run model tests
-              run: make test
+              run: |
+                make PYTHON=$PYTHON test
```

Comment:
In order to have both os tests share this run context, I'm trying to pass the os-specific python to make test. It seems to work, but I wonder if there was a better approach for setting this variable?

Reply:
I think the intent of the `setup-python` step was to do just that ... but I think we're in an interesting case here where we're trying to use the conda python and not just the system python, right?  If that's the case, does `setup-miniconda` set up a `python` binary that's accessible from other steps, much like `setup-python` sets up?

Reply:
I guess what I mean by this is that when we've used `setup-miniconda`, does that mean that when we call `python` in later steps, is that located within the conda environment?  If it is, it might be sufficient to call `make PYTHON=python`, or, if we need to be super specific about the path, we could specify the shell (bash is available on all OSes, I believe), and do a `make PYTHON=$(which python) test`.

Reply:
Good suggestions, it turns out a `setup-miniconda` system actually has both `python` and `python3` on the path. `python` being in the active conda env, and `python3` being `/usr/local/opt/python/bin/python3.7`. That fact, combined with the Makefile defaulting to `PYTHON=python3` on non-windows OS is what was causing problems. 

I pushed a change to get rid of the setting the environment variable, and calling `make PYTHON=python test`